File: Java/src/main/java/Solution301_400/Solution349.java
package Solution301_400;

import java.util.ArrayList;
import java.util.Arrays;
import java.util.HashSet;
import java.util.List;
import java.util.function.ToIntFunction;
import java.util.stream.Collectors;

/**
 * @author windows
 */
public class Solution349 {
    public int[] intersection(int[] nums1, int[] nums2) {
        HashSet<Integer> set = new HashSet<>();
        for (int i = 0; i < nums1.length; i++) {
            set.add(nums1[i]);
        }
        List<Integer> list = new ArrayList<>();
        for (int i = 0; i < nums2.length; i++) {
            if (set.contains(nums2[i])){
                list.add(nums2[i]);
                set.remove(nums2[i]);
            }
        }
        return Arrays.stream(list.toArray(new Integer[list.size()])).mapToInt(Integer::intValue).toArray();
    }


    public static void main(String[] args) {
        Solution349 solution349 = new Solution349();
    }
}

File: C++/src/Solution301_400/Solution349.cpp
//
// Created by windows on 2023/3/1.
//

#include "Solution349.h"
#include <unordered_set>

vector<int> Solution349::intersection(vector<int> &nums1, vector<int> &nums2) {
    unordered_set<int> hashset;
    unordered_set<int> res;
    vector<int> ans;
    for (int i = 0; i < nums1.size(); i++) {
        hashset.insert(nums1[i]);
    }
    for (int i = 0; i < nums2.size(); i++) {
        if (hashset.count(nums2[i])) {
            res.insert(nums2[i]);
        }
    }
    for (int a : res) {
        ans.push_back(a);
    }
    return ans;
}
